draft-ietf-protected-audience-key-value-services.md

```diff
@@ -0,0 +1,717 @@
+
+### Encryption {#encryption}
+
+The Key Value Service uses [HPKE] with the following configuration for encryption:
+
+-   HPKE KEM ID (Key encapsulation mechanisms):
+    0x0020 DHKEM(X25519, HKDF-SHA256), see {{Section 1 of HPKE}}
+-   HPKE KDF ID (key derivation functions): 0x0001 HKDF-SHA256, see {{Section 7.2 of HPKE}}
+-   HPKE AEAD ID (Authenticated Encryption with Associated Data): AES-256-GCM, see
+    {{Section 7.3 of HPKE}}
+
+### Message Framing {#framing}
+
+Before encryption and after decryption, the requests and responses have the following framing:
+
+| Byte     | 0         | 0             | 1 to 4   | 5 to Size+4       | Size+5 to end   |
+| -------- | --------- | ------------- | -------- | ----------------- | --------------- |
+| Bits     | 7-2       | 1-0           | \*       | \*                | \*              |
+| -------- | --------- | ------------- | -------- | ----------------- | --------------- |
+| Contents | Unused    | Compression   | Size     | Payload           | Padding         |
+
+The request/response is framed with this 5 byte header.
+
+The first byte is the format+compression byte. The lower 2 bits specify the format and compression ({{compression}}).
+The higher 6 bits are currently unused.
+
+The following 4 bytes are the length of the request message in network byte order.
+
+Padding is applied differently for request and response and will be discussed in the respective
+sections.
+
+### Format+compression byte {#compression}
+
+| Compression | Description                    |
+| :---------: | :----------------------------- |
+|    0x00     | [CBOR], no compression         |
+|    0x01     | [CBOR], compressed in [Brotli] |
+|    0x02     | [CBOR], compressed in [GZIP]   |
+|    0x03     | Reserved                       |
+
+Requests are always uncompressed so the Format+compression byte is 0x00.
+
+For responses, the byte value depends on the `acceptCompression` field
+(see {{request-schema}}) in the request and the server behavior.
+
+## Request Data {#request}
+
+Requests are not compressed and have a tree-like hierarchy:
+
+-   Each request contains one or more partitions. Each partition is a collection of keys that SHALL
+    be processed together by the service. Keys from separate partitions MUST NOT be processed together by the service.
+-   Each partition contains one or more key groups.
+    Each key group contains a list of `tags` set by the client.
+    Each key group contains a list of keys to be looked up in the service's internal datastore.
+-   Each partition has a unique identifier.
+    This allows the client to match the request partition with the corresponding
+    `partitionOutput` (see {{compression-group}}) in the response.
+-   Each partition has a compression group field.
+    Results of partitions belonging to the same compression group can be compressed together
+    in the response.
+    Different compression groups must be compressed separately (see {{compression-group}}).
```

Comment:
"Different compression groups must be compressed separately..." That's referring to the response, but from the context it could refer to the request.

So maybe something like:
"The responses for different compression groups will be compressed separately in the response (see {{compression-group}}). Compressing the different groups separately avoids leaking the similarity of responses for different groups."

Reply:
Done.